libkbfs/folder_branch_ops.go

```diff
@@ -1911,7 +1987,7 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		} else {
 			bid = fbo.bid
 		}
-		err := mdops.PutUnmerged(ctx, md, bid)
+		mdID, err = fbo.doPutWithCancellationCheck(ctx, lState, md, bid)
```

Comment:
Good call; that'd simplify `doPutWithCancellationCheck` (also rename) a lot as well!